File: Python3/프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dict = {}
    result = 1
    
    for i, j in clothes:
        if j in dict:
            dict[j] += 1
        else:
            dict[j] = 1
    
    for i in dict:
        result *= dict[i] + 1

    return result - 1
            

File: JavaScript/프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    const index = [];
    const result = [];
    
    for (let cloth of clothes) {
        const category = cloth[1];
        if (index.includes(category)) result[index.indexOf(category)] += 1;
        else {
            index.push(category);
            result.push(1);
        }
    }
    
    for (let value of result) answer *= (value + 1);
    
    return answer - 1;
}